build.rs

```diff
@@ -230,3 +500,203 @@ fn run_command_with_args<S>(command_name: S, args: &Vec<String>)
         panic!("{} execution failed", command_name.as_ref().to_str().unwrap());
     }
 }
+
+fn make_asm(source: &str, mut dst: PathBuf, target_info: &TargetInfo)
+            -> String {
+    let p = Path::new(source);
+    if p.extension().expect("File without extension").to_str() == Some("pl") {
+        dst.push(p.file_name().expect("File without filename??"));
+        dst.set_extension("S");
+        let r: String = dst.to_str().expect("Could not convert path").into();
+        let perl_include_changed = RING_PERL_INCLUDES.iter()
+            .any(|i| need_run(&Path::new(i), dst.as_path()));
+        if need_run(&p, dst.as_path()) || perl_include_changed {
+            let mut args = vec![source.to_owned()];
+            args.extend(target_info.perlasm_args()
+                .iter()
+                .map(|a| (*a).to_owned()));
+            args.push(r.clone());
+            run_command_with_args(&get_command("PERL_EXECUTABLE", "perl"),
+                                  &args);
+        }
+        r
+    } else {
+        p.to_str().expect("Could not convert path").into()
+    }
+}
+
+fn need_run(source: &Path, target: &Path) -> bool {
+    let s = std::fs::metadata(source);
+    let t = std::fs::metadata(target);
+    if s.is_err() || t.is_err() {
+        true
+    } else {
+        match (s.unwrap().modified(), t.unwrap().modified()) {
+            (Ok(s), Ok(t)) => s > t,
```

Comment:
Shouldn't this be `s >= t`? In particular, if you only have one-second-precision timestamps then it is very likely that you can't reliably know if the source was modified after the target, so you have to play it safe.